### PR TITLE
Updated data model in json file and added script to extract results.zip

### PR DESCRIPTION
**Pull request by Kenneth-Funes-TU-Delft**
- Created script to extract json files
- Modified analyse script to print into a file

### CODE REVIEW

File: preprocessing/speaking_status/annotations_v2/analyse_and_plot.py [hunk 1]
```diff
@@ -233,13 +233,19 @@ def main(
...
+    # 2. Run extract extract_json_files_from_results_zip_downloaded_from_covfee_admin.py
+    # 3. Run this script
+    # 4. Check the output in analysis_output.txt
+    import os
+    outfile = Path(os.path.join(os.path.dirname(__file__), "analysis_output.txt"))
+    with open(outfile, "w") as f:
+        import sys
+        sys.stdout = f
+
+        for path in Path(os.path.join(os.path.dirname(__file__), "json_files")).iterdir():
```
Comment (Era-Dorta) [inline preprocessing/speaking_status/annotations_v2/analyse_and_plot.py:246]: Ordering the files alphabetically makes the output file easier to navigate
```suggestion
        for path in sorted(Path(os.path.join(os.path.dirname(__file__), "json_files")).iterdir()):
```

**Review by Era-Dorta — APPROVED** [review]

File: preprocessing/speaking_status/annotations_v2/analyse_and_plot.py (hunk 1 above)
Comment (Kenneth-Funes-TU-Delft) [reply preprocessing/speaking_status/annotations_v2/analyse_and_plot.py]: Fixed in f477355b5e84adf87a0c0376259c119bd0f888b5

File: preprocessing/speaking_status/annotations_v2/extract_json_files_from_results_zip_downloaded_from_covfee_admin.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+os.makedirs(tmp_folder, exist_ok=True)
+
+print("Unzipping...")
+# Unzip the file
+with zipfile.ZipFile(COVFEE_ADMIN_PANEL_DOWNLOADED_RESULT_ZIP_FILE_PATH, 'r') as zip_ref:
+    zip_ref.extractall(tmp_folder)
+
+
+# Iterate over all json files in the tmp_folder
+for json_file in glob.glob(os.path.join(tmp_folder, '*.json')):
```
[the remark this answers is not in the document]
Comment (Kenneth-Funes-TU-Delft) [reply preprocessing/speaking_status/annotations_v2/extract_json_files_from_results_zip_downloaded_from_covfee_admin.py]: Fixed in f477355b5e84adf87a0c0376259c119bd0f888b5

File: preprocessing/speaking_status/annotations_v2/analyse_and_plot.py [hunk 3]
```diff
@@ -233,13 +233,19 @@ def main(
...
-        main([path], num_annotated_participants=48, do_plots=False, min_aggrement=0.7)
+    # 1. Download results.zip from the the covfee admin panel, assuming it's on ~/home/Downloads
+    # 2. Run extract extract_json_files_from_results_zip_downloaded_from_covfee_admin.py
+    # 3. Run this script
+    # 4. Check the output in analysis_output.txt
+    import os
+    outfile = Path(os.path.join(os.path.dirname(__file__), "analysis_output.txt"))
+    with open(outfile, "w") as f:
+        import sys
+        sys.stdout = f
```
[the remark this answers is not in the document]
Comment (Kenneth-Funes-TU-Delft) [reply preprocessing/speaking_status/annotations_v2/analyse_and_plot.py]: Discussed f2f. We'll leave as is, i.e. no assumption on command line use, except for improvements on code clarity

File: preprocessing/speaking_status/annotations_v2/analyse_and_plot.py [hunk 4]
```diff
@@ -233,13 +233,19 @@ def main(
...
-    # ]
-
-    for path in Path("/home/era/code/covfee-repos/covfee_databases/json_batch_03/").iterdir():
-        main([path], num_annotated_participants=48, do_plots=False, min_aggrement=0.7)
+    # 1. Download results.zip from the the covfee admin panel, assuming it's on ~/home/Downloads
+    # 2. Run extract extract_json_files_from_results_zip_downloaded_from_covfee_admin.py
+    # 3. Run this script
+    # 4. Check the output in analysis_output.txt
+    import os
+    outfile = Path(os.path.join(os.path.dirname(__file__), "analysis_output.txt"))
```
[the remark this answers is not in the document]
Comment (Kenneth-Funes-TU-Delft) [reply preprocessing/speaking_status/annotations_v2/analyse_and_plot.py]: Added batch filtering in f477355b5e84adf87a0c0376259c119bd0f888b5

File: preprocessing/speaking_status/annotations_v2/extract_json_files_from_results_zip_downloaded_from_covfee_admin.py [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+COVFEE_ADMIN_PANEL_DOWNLOADED_RESULT_ZIP_FILE_PATH: str = os.path.join(home_dir, "Downloads", "results.zip")
+OUTPUT_FOLDER_PATH_FOR_JSON_FILES: str = os.path.join(os.path.dirname(__file__), "json_files")
+
+################################################################################
+# Execution below - do not modify
+
+if not os.path.exists(COVFEE_ADMIN_PANEL_DOWNLOADED_RESULT_ZIP_FILE_PATH):
+    print(f"The results zip file does not exist: {COVFEE_ADMIN_PANEL_DOWNLOADED_RESULT_ZIP_FILE_PATH}")
+    exit()
+
```
[the remark this answers is not in the document]
Comment (Kenneth-Funes-TU-Delft) [reply preprocessing/speaking_status/annotations_v2/extract_json_files_from_results_zip_downloaded_from_covfee_admin.py:23]: Is this really needed ? if yes, what should happen to the old json files ? All of them be deleted ? be replaced for the ones found in results.zip ?

**Review by Era-Dorta — APPROVED** [review]

**Pull request merged**